src/main/java/com/google/maps/model/QueryAutocompletePrediction.java

```diff
@@ -0,0 +1,48 @@
+package com.google.maps.model;
+
+public class QueryAutocompletePrediction {
+
+  /**
+   * Description of the matched prediction.
+   */
+  public String description;
+
+  /**
+   * Deprecated field. Do not use.
+   *
+   * @deprecated This field is deprecated. Please see
+   * <a href="https://developers.google.com/places/web-service/query#deprecation">the deprecation notice</a>.
+   */
+  public String id;
+
+  /**
+   * Deprecated field. Do not use.
+   *
+   * @deprecated This field is deprecated. Please see
+   * <a href="https://developers.google.com/places/web-service/query#deprecation">the deprecation notice</a>.
+   */
+  public String reference;
+
+  /**
+   * The Place ID of the place.
+   */
+  public String placeId;
+
+  public String types[];
+
+  // TODO(brettmorgan): Document this on https://developers.google.com/places/web-service/query
+  public static class MatchedSubstring {
```

Comment:
sure, that's fine. I wasn't sure of how there were constructed, at this
point in reading.

Let's talk tomorrow about the possibility of using something like
ValueType, though.

On Sun, 27 Sep 2015, 21:15 Brett Morgan <notifications@github.com> wrote:

> In src/main/java/com/google/maps/model/QueryAutocompletePrediction.java
> <https://github.com/googlemaps/google-maps-services-java/pull/101#discussion_r40503451>
> :
>
> > +   * Deprecated field. Do not use.
> > +   *
> > +   * @deprecated This field is deprecated. Please see
> > +   * <a href="https://developers.google.com/places/web-service/query#deprecation">the deprecation notice</a>.
> > +   */
> > +  public String reference;
> > +
> > +  /**
> > +   * The Place ID of the place.
> > +   */
> > +  public String placeId;
> > +
> > +  public String types[];
> > +
> > +  // TODO(brettmorgan): Document this on https://developers.google.com/places/web-service/query
> > +  public static class MatchedSubstring {
>
> That'd require putting in a GSON mapper. I'm unsure of the gain here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlemaps/google-maps-services-java/pull/101/files#r40503451>
> .
>